### PR TITLE
Make Select accept array as options

### PR DESCRIPTION
**Pull request by qimyi**

### CODE REVIEW

**Comment by jameslnewell** [comment]
Hey @qimyi thanks for the PR! 

What's the use case for this PR? Is `{value: '...', text: '...'}` an accepted standard anywhere? I ask because I don't want to have to cater to each team's preferred format for `options` (the property names will vary based on the data source so there'll usually need to be some kind of transformation anyway and `["NSW", "VIC", ...]` is another commonly used array format).

Is there any reason you can't do something like:

``` javascript

const convertOptionsArrayToObject = optionsArray => optionsArray.reduce((options, item) => {
  ...options, 
  [item.value]: item.text
}, {});

const fruits = [
  {
    value: 'APL',
    text: 'Apple'
  },
  {
    value: 'BAN',
    text: 'Banana'
  }
];

<Form.Select options={convertOptionsArrayToObject(fruits)}/>

```

**Comment by qimyi** [comment]
Hey @jameslnewell, 

The concern is that the order of the Object is not guaranteed but in an Array, it is. Sorry for not making that clear earlier.

Thanks

**Comment by iknowcss** [comment]
Here's an example:

```
var fruits = { apple: "Apple", kiwi: "Kiwi" }

Object.keys(fruits)
// [ 'apple', 'kiwi' ]

fruits.banana = "Banana"
// 'Banana'

Object.keys(fruits)
// [ 'apple', 'kiwi', 'banana' ]
```

There's no way to make banana come before kiwi unless there's sort logic in the component somewhere

**Comment by jameslnewell** [comment]
Sweet, good use case.

Can we use `label` instead of `text`... seems to be a lot more libraries already using this convention:
- https://github.com/JedWatson/react-select#usage (I think this is one of the more popular ones)
- http://furqanzafar.github.io/react-selectize/#/
- https://github.com/alexkuz/react-input-enhancements#dropdown
- https://github.com/bharani91/react-select-popover

File: lib/fields/Select.jsx [hunk 1]
```diff
@@ -9,23 +10,30 @@ import classNames from 'classnames';
...
   );
-
+    
 }
 
 Select.propTypes = {
   className: React.PropTypes.string,
   placeholder: React.PropTypes.string,
-  options: React.PropTypes.object.isRequired
+  options: React.PropTypes.any.isRequired
```
Comment (jameslnewell) [inline lib/fields/Select.jsx]: can we use 

``` javascript
React.PropTypes.oneOfType([
  React.PropTypes.arrayOf(
    React.PropTypes.shape({
      value:  React.PropTypes.string,
      label:   React.PropTypes.string
    })
  ),
  React.PropTypes.objectOf(React.PropTypes.string)
])
```

**Comment by qimyi** [comment]
That's fine, will do

**Comment by iknowcss** [comment]
Ah I knew there must be something like `oneOfType`! We couldn't get Google to tell us about it. Thanks :+1:

**Comment by jameslnewell** [comment]
This cheatsheet is amaze balls :exclamation::exclamation::exclamation:  http://ricostacruz.com/cheatsheets/react.html

**Comment by jameslnewell** [comment]
Published in `@nib-components/react-form@0.7.0`

**Pull request merged**